cpp-compile.Dockerfile

```diff
@@ -10,56 +10,59 @@ ENV OS_DEP "\
       ca-certificates \
       curl \
       gnupg \
+      cmake-curses-gui \
```

Comment:
I do not think we will need this. If a developer wants to, then they can just run the cmake gui from their host to see the possible options. Compiling cmake with ccmake further down below can still be done. It does not require any additional dependencies anyways